packages/shared/src/hooks/useOrganizationList.tsx

```diff
@@ -1,39 +1,191 @@
 import type {
+  ClerkPaginationResponse,
   CreateOrganizationParams,
+  GetUserOrganizationInvitations,
   OrganizationMembershipResource,
   OrganizationResource,
   SetActive,
+  UserOrganizationInvitationResource,
+  UserResource,
 } from '@clerk/types';
+import { useMemo } from 'react';
+import useSWR from 'swr';
+import useSWRInfinite from 'swr/infinite';
 
 import { useClerkInstanceContext, useUserContext } from './contexts';
 
+type UseOrganizationListParams = {
+  userInvitations?:
+    | true
+    | (GetUserOrganizationInvitations & {
+        aggregate: boolean;
+      });
+};
+
 type OrganizationList = ReturnType<typeof createOrganizationList>;
 
 type UseOrganizationListReturn =
-  | { isLoaded: false; organizationList: undefined; createOrganization: undefined; setActive: undefined }
   | {
-      isLoaded: true;
+      isLoaded: false;
+      organizationList: undefined;
+      createOrganization: undefined;
+      setActive: undefined;
+      userInvitations: {
+        data: undefined;
+        count: number;
+        isLoading: false;
+        isValidating: false;
+        isError: false;
+        size: undefined;
+        setSize: undefined;
+      };
+    }
+  | {
+      isLoaded: boolean;
       organizationList: OrganizationList;
       createOrganization: (params: CreateOrganizationParams) => Promise<OrganizationResource>;
       setActive: SetActive;
+      userInvitations: {
+        data: UserOrganizationInvitationResource[];
+        count: number;
+        isLoading: boolean;
+        isValidating: boolean;
+        isError: boolean;
+        size: number | undefined;
+        setSize:
+          | ((
+              size: number | ((_size: number) => number),
+            ) => Promise<(ClerkPaginationResponse<UserOrganizationInvitationResource> | undefined)[] | undefined>)
+          | undefined;
```

Comment:
Can we probably use the `Dispatch<SetStateAction<...>>` interface from React?

Reply:
This is the type from SWR, `Dispatch<SetStateAction<...>>` returns void, right ?

Probably this will change, as I am working on a more uniform exposed API

Reply:
A type abstraction would be nice, but yes you're right :)